src/reference.rs

```diff
@@ -10,6 +10,36 @@ pub enum ReferenceOr<T> {
     Item(T),
 }
 
+impl<'de, T> Deserialize<'de> for ReferenceOr<T>
```

Comment:
maybe it's worth having a comment explaining why we have this custom `Deserialize` implementation, and how it helps us get nicer errors than just deriving `Deserialize`?